### ISSUE TITLE
Differences in results

### ISSUE THREAD

**Issue by zbuk78** [ddurieux/glpi_app_grafana#24]
Hi David! I vave a problem setting up a Singlestat panel so that it shows me the exact same number of items found in glpi. Here's an example:
https://my.glpi.address/glpi/front/computer.php?is_deleted=0&criteria[0][field]=5150&criteria[0][searchtype]=morethan&_select_criteria[0][value]=-4DAY&criteria[0][value]=-4DAY&search=Szukaj&itemtype=Computer&start=0&_glpi_csrf_token=292d836fc52c10bc8d1a418730329b84
This search returns the number 1108. Then I paste the link to a grafana query field, check count elements checkbox, in third tab (Options) I change Stats to Current, which gives me result of 1 item. When I chenge Stats to Total it returns 1090. I couldn't find any explenation on how to use this plugin properly to get the same results as in glpi.
Thanks for Your help.

**Comment by ddurieux** [comment]
Be carefull, because you have the time restriction in grafana, it can modify the counter

**Comment by zbuk78** [comment]
Yes, changing date range I get different results, but still none of them (even with last 50 years) is the same results in glpi and grafana. What's more, when I run Query Inspector, then I have three different results:
glpi - 1176
Query inspector returns 1099
and on grafana's singlestat I have 1077 machines.
I realize that may not be a bug and I have misconfigured something, but I'm just not able to get results that would be exactly the same.

**Comment by ddurieux** [comment]
Ok, I will try to check / reproduce

**Comment by zbuk78** [comment]
One more strange thing I've noticed is that some queries show good results in Query Inspector and after a while the url changes itself for one searching tickets (even though I paste the url to look for ie. virtual machines) and the request ends up with something like that:

request:Object
method:"GET"
url:"api/datasources/proxy/8/search/ticket?is_deleted=0&criteria[0][field]=12&criteria[0][searchtype]=equals&criteria[0][value]=all&search=Rechercher&itemtype=Ticket&start=0&criteria[1][link]=AND&criteria[1][field]=15&criteria[1][searchtype]=morethan&_select_criteria[1][value]=0&_criteria[1][value]=-49624586&criteria[1][value]=1968-06-05 15:23:34&criteria[2][link]=AND&criteria[2][field]=15&criteria[2][searchtype]=lessthan&_select_criteria[2][value]=0&_criteria[2][value]=1528212214&criteria[2][value]=2018-06-05 15:23:34&forcedisplay[0]=0&giveItems=true&forcedisplay[1]=1&forcedisplay[3]=3&forcedisplay[13]=13&forcedisplay[4]=4&forcedisplay[5]=5&forcedisplay[15]=15&range=2000-2399"
response:Object
totalcount:42630
count:400
sort:1
order:"ASC"

Before that result I have this one:
request:Object
method:"GET"
url:"api/datasources/proxy/8/search/computer?is_deleted=0&criteria[0][field]=5150&criteria[0][searchtype]=morethan&_select_criteria[0][value]=-1WEEK&criteria[0][value]=-1WEEK&criteria[1][link]=AND&criteria[1][field]=40&criteria[1][searchtype]=equals&criteria[1][value]=26&criteria[2][link]=OR&criteria[2][field]=40&criteria[2][searchtype]=equals&criteria[2][value]=15&search=Szukaj&itemtype=Computer&start=0&_glpi_csrf_token=8209811f9da600ba98e78b98dbcc0745&criteria[3][link]=AND&criteria[3][field]=5150&criteria[3][searchtype]=morethan&_select_criteria[3][value]=0&_criteria[3][value]=1370446314&criteria[3][value]=2013-06-05 15:31:54&criteria[4][link]=AND&criteria[4][field]=5150&criteria[4][searchtype]=lessthan&_select_criteria[4][value]=0&_criteria[4][value]=1528212714&criteria[4][value]=2018-06-05 15:31:54&forcedisplay[0]=0&range=0-399"
response:Object
totalcount:337
count:337
sort:1
which is the exact number returned by glpi, but the number displayed on singlestat is 322.

**Comment by yoshi314** [comment]
also noticing this, simple query for computers that have not been updated with fusioninventory since last week is off by ~100

**Comment by eprogres** [comment]
Hi David,
Did you reproduce the problem? Any progress with this issue?

**Comment by ddurieux** [comment]
I planned to work on grafana this friday

**Comment by eprogres** [comment]
![grafana_screen](https://user-images.githubusercontent.com/15325560/47705289-64a1e680-dc26-11e8-85bf-407f28073188.png)


nothing better than screenshot 
Query inspector in line totalcount:1236 shows correct value, same number in glpi
but in indicator we got **1095** which is wrong.  Time range in query is correct (in example last 7 days)

we try to count returned objects - checked option "Count elements"

**Comment by ddurieux** [comment]
You define 'total' in 'stat' field in tab 'options' ?

**Comment by eprogres** [comment]
![total](https://user-images.githubusercontent.com/15325560/47778630-93d45880-dcf7-11e8-970d-5c6758337f79.png)

Yes we did
still wrong number

**Comment by eprogres** [comment]
[dashboard.txt](https://github.com/ddurieux/glpi_app_grafana/files/2533625/dashboard.txt)

**Comment by ddurieux** [comment]
Hello, can you try with this patch: https://github.com/ddurieux/glpi_app_grafana/commit/21d0b1a81911937ae6f6f6f2790a7c0f0fe4b9ac

**Comment by eprogres** [comment]
![zrzut ekranu z 2018-11-05 14-43-36](https://user-images.githubusercontent.com/15325560/48001607-5b72b700-e109-11e8-82ed-b657496b7b13.png)

still discrepancy in results :(

**Comment by Marsnet100** [comment]
is there any progress ?  this plugin would be very valuable for us :|

**Comment by ddurieux** [comment]
Not had yet time to check, can you see if you have already the problem with version 1.3 ?

**Comment by kowalskimn** [comment]
We're still observing a constant offset between query inspector results and displayed results. 

The difference varies every day, but seems fixed when changing time ranges in grafana.
